api/sys_mounts.go

```diff
@@ -186,6 +186,26 @@ func (c *Sys) RemountStatusWithContext(ctx context.Context, migrationID string)
 	return &result, err
 }
 
+func (c *Sys) TuneAuth(path string, config MountConfigInput) error {
+	return c.TuneAuthWithContext(context.Background(), path, config)
+}
+
+func (c *Sys) TuneAuthWithContext(ctx context.Context, path string, config MountConfigInput) error {
+	ctx, cancelFunc := c.c.withConfiguredTimeout(ctx)
+	defer cancelFunc()
+
+	r := c.c.NewRequest(http.MethodPost, fmt.Sprintf("/v1/sys/auth/%s/tune", path))
```

Comment:
Looks like this is not the same path that was previously being validated:
`/v1/sys/auth/<auth-path>/tune`, v. `/v1/sys/mounts/auth/<auth-path>/tune`. Is this intentional?

I'm also curious as to why we need a new api for these. Can't we just continue using the `TuneMount` api?

Reply:
Yes, we can tune an auth mount using auth/tune (needs sudo capability) or using mounts/auth/tune. On the cli, auth tune command uses mounts/auth/<path>/tune api. This did not cause any problems until now because all the fields for auth tune could be changed using mounts/auth/<path>tune. Now, that user_lockout_config can be changed only with sudo capability, I had to change this. If we left this as it is, then it would use the mounts/auth/<path>/tune api and would have errored out there. Therefore, I changed the api call on the cli as well.

Reply:
@akshya96 and I discussed out-of-band and it looks like the `auth tune` cli was not enforcing `sudo` permissions previously (it was using `/sys/mounts/auth` under the hood and bypassing the `/sys/auth` sudo path check. This appears to be a bugfix. We're going to look into adding a test to prevent future regression.

Reply:
Added a test to check sudo capability for  vault auth tune

Reply:
Oh, sorry I missed this initially. Would you mind moving this to `sys_auth.go`?